metaflow/plugins/aws/batch/batch.py

```diff
@@ -62,6 +62,11 @@ def _job_name(self, user, flow_name, run_id, step_name, task_id, retry_count):
             retry_count=retry_count,
         )
 
+        if len(job_name) > 128:
+            raise BatchException('Cant run a batch job with a name exceeding 128 characters. Try reducing your flow or step names length')
```

Comment:
Maybe we can reword this error as - `Unable to launch AWS Batch job - job name exceeds 128 characters. Shorten your flow and/or step name by x characters.` Currently, the user has no idea how much to reduce the length by.